hw07_file_copying/copy.go

```
@@ -2,14 +2,68 @@ package main
 
 import (
 	"errors"
+	"io"
+	"os"
 )
 
 var (
 	ErrUnsupportedFile       = errors.New("unsupported file")
 	ErrOffsetExceedsFileSize = errors.New("offset exceeds file size")
```

не нахожу ни одной проверки на unsupported файлы (urand тот же, proc чтение из папок и прочее), также не нахожу где мы используем вторую ошибку. Вообще рекомендую fail fast approach при создании подобных  утилит, где мы сначала проверяем что входные данные у нас соответствуют необходимым, и только после этого начинаем выполнение бизнес логики. Если же есть какое то не соответствие то рекомендую возвращать правильные коды ошибок / ошибки юзеру чтобы было понятно что нужно поправить